### ISSUE TITLE
Rewrite e2e tests for `inverted_refund` and `refund`

### ISSUE THREAD

**Issue by luckysori** [comit-network/comit-rs#1594]
## Problem

Only one e2e test (btc-eth happy path) was rewritten in PR #1584 following the new design outlined in [spike 11](https://github.com/comit-network/spikes/blob/master/0011-revise-e2e-test-framework.adoc#the-ideal-test).

## Goal

Use the new design for all our e2e tests.

## Proposal

Do for albtc-eth tests: `inverted_refund.ts`, `refund.ts`.

**Comment by bonomat** [comment]
The PR in #1709 converts `refund` and `inverted_refund` to the new format. However, for over/under-funding our SDK is too correct, i.e. it does not let you misuse the `fund` function (which is good).

In order to test this nevertheless and not have to write the action interpretation again just for the e2e tests, I would propose a change in the sdk and make the functions public: 

https://github.com/comit-network/comit-js-sdk/blob/563cbff7df763d71f5496d00548fc7be6b9ab0ff/src/swap.ts#L49-L57
and 
https://github.com/comit-network/comit-js-sdk/blob/563cbff7df763d71f5496d00548fc7be6b9ab0ff/src/swap.ts#L59

@comit-network/rust-devs any thoughts on that?

**Comment by thomaseizinger** [comment]
Hmm, interesting insight. I'd hesitate a bit in extending the public API surface too much. Can we maybe achieve the same functionality by adding more parameters to the `fund` etc actions? Just in case users want to override something?

> it does not let you misuse the fund function (which is good).

Maybe these tests should just use a lower level of the SDK and do things "manually"?

**Comment by D4nte** [comment]
Can you bypass the fact that they are not exported by any chance? (I am guessing it's a JS thing and not a TS thing you could bypass but better try).

**Comment by thomaseizinger** [comment]
> Can you bypass the fact that they are not exported by any chance? (I am guessing it's a JS thing and not a TS thing you could bypass but better try).

I don't think you can because I believe that is actually enforced on the module level, even in JavaScript.

**Comment by bonomat** [comment]
> Can we maybe achieve the same functionality by adding more parameters to the fund etc actions? Just in case users want to override something?

I would rather not allow passing something into the `fund` method. Because at the end we want the sdk to be foolproof.

Providing some low level APIs which enable the user to have more fine grained control over the actions sounds reasonable. 

But let's not go this path for now and rather rethink how/what we want to achieve with the e2e tests. 

Would you be fine with closing this ticket once PR #1709 is merged and opening a new one to deal with the invalid funding tests?

**Comment by thomaseizinger** [comment]
> Would you be fine with closing this ticket once PR #1709 is merged and opening a new one to deal with the invalid funding tests?

Sounds good to me :)

**Comment by D4nte** [comment]
@bonomat  Just make current work work and create new issue for remaining tests

**Comment by bonomat** [comment]
Fixed in #1771
